core/domain/exp_jobs_one_off_test.py

```diff
@@ -1834,3 +1838,426 @@ def test_no_action_is_performed_for_deleted_exploration(self):
+                job_id)
+            )
+        expected_output = [('[u\'State schema version is '
+                            'not the minimum version expected, 31\', '
+                            '[u\'EXP_ID: exp_id0, '
+                            'STATE_SCHEMA_VERSION: 30\']]')]
+        self.assertEqual(actual_output, expected_output)
+
+    def test_duration_secs_is_updated_when_job_is_complete(self):
+        exploration = exp_domain.Exploration.create_default_exploration(
+            self.VALID_EXP_ID, title='Exploration Title 1', category='category')
+        exploration.add_states(['State1'])
+        state1 = exploration.states['State1']
+        recorded_voiceovers_dict = {
+            'voiceovers_mapping': {
+                'content': {
+                    'en': {
+                        'filename': self.TEST_AUDIO_FILE_MP3,
+                        'file_size_bytes': 123,
+                        'needs_update': True,
+                        'duration_secs': 0.0
+                    },
+                    'hi': {
+                        'filename': self.TEST_AUDIO_FILE_MP3,
+                        'file_size_bytes': 1234,
+                        'needs_update': False,
+                        'duration_secs': 0.0
+                    }
+                },
+                'default_outcome': {}
+            }
+        }
+
+        # Get the test audio file and prepare for storage.
+        with python_utils.open_file(
+            os.path.join(feconf.TESTS_DATA_DIR, self.TEST_AUDIO_FILE_MP3),
+            mode='rb', encoding=None) as f:
+            raw_audio = f.read()
+
+        mimetype = 'audio/mp3'
+
+        # Save the raw audio into the filesystem.
+        file_system_class = fs_services.get_entity_file_system_class()
+        fs = fs_domain.AbstractFileSystem(file_system_class(
+            feconf.ENTITY_TYPE_EXPLORATION, self.VALID_EXP_ID))
+        fs.commit(
+            '%s/%s' % (self._FILENAME_PREFIX, self.TEST_AUDIO_FILE_MP3),
+            raw_audio, mimetype=mimetype)
+
+        # Save the recorded_voiceovers.
+        state1.update_recorded_voiceovers(
+            state_domain.RecordedVoiceovers.from_dict(recorded_voiceovers_dict))
+        exp_services.save_new_exploration(self.user_a_id, exploration)
+
+        # Run the job.
+        job_id = (
+            exp_jobs_one_off.VoiceoverDurationSecondsOneOffJob.create_new())
+        exp_jobs_one_off.VoiceoverDurationSecondsOneOffJob.enqueue(job_id)
+        self.process_and_flush_pending_tasks()
+
+        actual_output = (
+            exp_jobs_one_off.VoiceoverDurationSecondsOneOffJob.get_output(
+                job_id)
+            )
+        # Validate job ran successfully.
+        expected_output = [('[u\'SUCCESS_AUDIO_CHANGED\', [u\'EXP_ID: exp_id0, '
+                            'AUDIO_DURATIONS_CHANGED: 2\']]')]
+        self.assertEqual(actual_output, expected_output)
+
+
```

Comment:
Drop one newline from the "double newline", here and elsewhere.

This was mentioned in a previous review. In general, when a review comment says "and elsewhere", please make sure to scan your changes for other instances of the same issue, and fix them. Thanks!

Reply:
Sorry I missed thought I got all of them.

Reply:
ok done.